src/Data/Auth/AuthProvider.tsx

```diff
@@ -0,0 +1,27 @@
+import React from "react";
```

Comment:
I would like to see at least a comment (and maybe also better naming) to make it clear that this is the factory function

Reply:
Added